hpx/lcos/detail/future_data.hpp

```diff
@@ -296,6 +297,20 @@ namespace detail
             error_code& = throws) = 0;
         virtual std::exception_ptr get_exception_ptr() const = 0;
 
+        virtual std::string const& get_registered_name() const
```

Comment:
Is this really required to be part of the abstract base? We only need it for the id_type specialization, right?

Reply:
Yes, it's only needed for that. The alternative is to upcast the shared state - which I'd like to avoid.